app/src/main/java/com/feedhenry/securenativeandroidtemplate/MainActivity.java

```diff
@@ -54,7 +111,210 @@ public void onBackPressed() {
     }
 
     /**
+     * Create the config for the initial Keycloak auth request to get a temporary token and create an intent to handle the response
+     */
+    public void performKeycloakAuthentication () {
```

Comment:
I think it will make things easier to read in this class if we move the implementation of authentication process to a different class. It will also allow us to provider alternative implementation if we want, without changing the main activity.